eogrow/utils/testing.py

```diff
@@ -168,10 +168,14 @@ def _calculate_tiff_stats(tiff_filename: str, config: StatCalcConfig) -> JsonDic
 def _calculate_vector_stats(gdf: gpd.GeoDataFrame, config: StatCalcConfig) -> JsonDict:
     """Calculates statistics over a vector GeoDataFrame"""
 
-    def _rounder(x: float, y: float) -> tuple[float, float]:
+    def _rounder(coords: tuple[float, float]) -> tuple[float, float]:
+        x, y = coords
         return _prepare_value(x, np.float64), _prepare_value(y, np.float64)
 
-    gdf.geometry = gdf.geometry.apply(lambda geometry: shapely.ops.transform(_rounder, geometry))
+    def _get_coords(geom: Polygon | MultiPolygon | Any) -> list[tuple[float, float]] | None:
```

Comment:
I would rename this function to `_get_coords_sample` to more explicitly show that only a subset of points is taken.